src/Processors/QueryPlan/AggregatingStep.cpp

```diff
@@ -333,6 +345,13 @@ void AggregatingStep::transformPipeline(QueryPipelineBuilder & pipeline, const B
 
     if (!sort_description_for_merging.empty())
     {
+        if (!inputDataSuitablyOrdered(input_streams.front(), sort_description_for_merging))
```

Comment:
It'd be nicer to fix sort description instead of providing the crutch. I guess`ActionsDAG::isSortingPreserved()` doesn't handle this case well. Up to you to decide, but in case you leave it as is, let's create an issue as follow-up to remove the crutch later